### PR TITLE
External access snowpark blog with eqt

### PR DESCRIPTION
**Pull request by ernestoongaro**
Creating a new blog post about external access auth, https://github.com/dbt-labs/docs-internal/pull/621

### CODE REVIEW

**Review by jasnonaz — APPROVED** [review]
Super interesting use case, thanks for writing.

File: website/blog/2024-06-12-putting-your-dag-on-the-internet.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+To start, we need to set up a network rule (Snowflake instructions [here](https://docs.snowflake.com/en/user-guide/network-rules)) to allow access to the external API. Specifically, we'll create an egress rule to permit Snowflake to communicate with api.carbonintensity.org.
+
+Next, in order to access network locations external to Snowflake an external access integration has to be defined first and then referenced within a dbt Python model (Snowflake overview [here](https://docs.snowflake.com/en/developer-guide/external-network-access/external-network-access-overview)). This API is open, if it required an API key, you’d do something similar for secrets (Snowflake overview [here](https://docs.snowflake.com/en/user-guide/api-authentication)).
```
Comment (mirnawong1) [inline website/blog/2024-06-12-putting-your-dag-on-the-internet.md]:
```suggestion
Next, to access network locations outside of Snowflake, you need to define an external access integration first and reference it within a dbt Python model. You can find an overview of Snowflake's external network access [here](https://docs.snowflake.com/en/developer-guide/external-network-access/external-network-access-overview).

This API is open and if it requires an API key, handle it similarly to managing secrets. More information on API authentication in Snowflake is available [here](https://docs.snowflake.com/en/user-guide/api-authentication).
```

File: website/blog/2024-06-12-putting-your-dag-on-the-internet.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+To start, we need to set up a network rule (Snowflake instructions [here](https://docs.snowflake.com/en/user-guide/network-rules)) to allow access to the external API. Specifically, we'll create an egress rule to permit Snowflake to communicate with api.carbonintensity.org.
+
+Next, to access network locations outside of Snowflake, you need to define an external access integration first and reference it within a dbt Python model. You can find an overview of Snowflake's external network access [here](https://docs.snowflake.com/en/developer-guide/external-network-access/external-network-access-overview).
+
+This API is open and if it requires an API key, handle it similarly to managing secrets. More information on API authentication in Snowflake is available [here](https://docs.snowflake.com/en/user-guide/api-authentication).
+
+For simplicity’s sake, we will show how to create them using pre-hooks in a model configuration yml file (dbt pre-hooks overview [here](/reference/resource-configs/pre-hook-post-hook)):
```
Comment (mirnawong1) [inline website/blog/2024-06-12-putting-your-dag-on-the-internet.md]:
```suggestion
For simplicity’s sake, we will show how to create them using [pre-hooks](/reference/resource-configs/pre-hook-post-hook) in a model configuration yml file:
```

File: website/blog/2024-06-12-putting-your-dag-on-the-internet.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+from raw,
+    lateral flatten(input => raw.carbon_intensity_json:data)
+```
+
+
+The result is a model that will keep track of dbt invocations, and the current UK carbon intensity levels.
+
+<Lightbox src="/img/blog/2024-06-12-putting-your-dag-on-the-internet/image1.png" title="Preview in dbt Cloud IDE of output" />
+
+## dbt Best Practices
```
Comment (mirnawong1) [inline website/blog/2024-06-12-putting-your-dag-on-the-internet.md]:
```suggestion
## dbt best practices
```

File: website/blog/2024-06-12-putting-your-dag-on-the-internet.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+Traditionally dbt is the T in ELT (dbt overview [here](https://docs.getdbt.com/terms/elt)), and this functionality unlocks brand new EL capabilities for which best practices do not yet exist. What’s clear is that EL workloads should be separated from T workloads, perhaps in a different modeling layer. Note also that unless using incremental models, your historical data can easily be deleted. dbt has seen a lot of use cases for this, including this AI example as outlined in this external [engineering blog post](https://klimmy.hashnode.dev/enhancing-your-dbt-project-with-large-language-models). 
+
+**A few words about the power of Commercial Open Source Software**
+
+In order to get this functionality shipped quickly, EQT opened a pull request, Snowflake helped with some problems we had with CI and a member of dbt Labs helped write the tests and merge the code in!  
+
+dbt now features this functionality in dbt 1.8+ or on “keep on latest version” option of dbt Cloud (dbt overview [here](/docs/dbt-versions/upgrade-dbt-version-in-cloud#keep-on-latest-version)). dbt staff and community members would love to chat more about it in #db-snowflake
```
Comment (mirnawong1) [inline website/blog/2024-06-12-putting-your-dag-on-the-internet.md]:
```suggestion
dbt now features this functionality in dbt 1.8+ or on “Keep on latest version” option of dbt Cloud (dbt overview [here](/docs/dbt-versions/upgrade-dbt-version-in-cloud#keep-on-latest-version)). 

dbt Labs staff and community members would love to chat more about it in the [#db-snowflake](https://getdbt.slack.com/archives/CJN7XRF1B) slack channel.
```

**Review by mirnawong1 — APPROVED** [review]
looks great, some very tiny tweaks (like adding the slack channel link) and have approved for you @ernestoongaro !

File: website/blog/2024-06-12-putting-your-dag-on-the-internet.md [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+title: Putting Your DAG on the internet
+description: "Use dbt and Snowflake's external access integrations to allow Snowflake Python models access the internet."
+slug: dag-on-the-internet
+
+authors: [ernesto_ongaro, sebastian_stan, filip_byrén]
+
+tags: [analytics craft, APIs, data ecosystem]
+hide_table_of_contents: false
+
+date: 2024-06-07
```
Comment (mirnawong1) [inline website/blog/2024-06-12-putting-your-dag-on-the-internet.md]: @ernestoongaro i'd recommend updating this to the merge/publish date

```suggestion
date: 2024-06-13
```

File: website/blog/2024-06-12-putting-your-dag-on-the-internet.md [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+title: Putting Your DAG on the internet
+description: "Use dbt and Snowflake's external access integrations to allow Snowflake Python models access the internet."
+slug: dag-on-the-internet
+
+authors: [ernesto_ongaro, sebastian_stan, filip_byrén]
+
+tags: [analytics craft, APIs, data ecosystem]
+hide_table_of_contents: false
+
+date: 2024-06-13
```
Comment (mirnawong1) [inline website/blog/2024-06-12-putting-your-dag-on-the-internet.md]:
```suggestion
date: 2024-06-14
```

**Pull request merged**